.github/workflows/testing.yml

```diff
@@ -32,6 +32,7 @@ jobs:
 
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: true
```

Comment:
IIUC `fail-fast: true` will reduce the amount of signal you get from a CI run. e.g. if a job running on 3.9-3.12 fails on 3.9 first, you won't know if it also would have failed on 3.10, 3.11, etc. is this intended?

Reply:
I mostly added this for my ease of figuring out why tests were failing initially - I can revert if you want